### ISSUE TITLE
ORM: Add "foreign" key index

### ISSUE THREAD

**Issue by alpe** [regen-network/cosmos-modules#23]
TlDr;: I would like to replace the following 2 Gets on tales by an index.
```go
func (k Keeper) GetGroupByGroupAccount(ctx sdk.Context, address sdk.AccAddress) (GroupMetadata, error) {
	var obj GroupAccountMetadataBase
	if err := k.groupAccountTable.GetOne(ctx, address.Bytes(), &obj); err != nil {
		return GroupMetadata{}, errors.Wrap(err, "load group account")
	}
	return k.GetGroup(ctx, obj.Group)
}
```
---
Scenario:
I have Groups and GroupAccounts.
```protobuf
message GroupAccount {
    bytes address = 1 ;
    uint64 group = 2 ;
}
```
When I create a new proposal we refer to a `GroupAccount` in the payload. To store the group version with the proposal, I would need to load the group via reference from GroupAccount. 
Instead of the 2 steps to load `GroupAccount` by address and then load `Group` by `GroupAccount.Group` we could introduce a new index type that points to the `Group` entity from an `GroupAccount`.

All our current indexes are work with the `RowGetter` and type of the table builder where they are registered. For this foreign key index we would need the `Group` `RowGetter` and type but register on the `GroupAccounts` builder with and raw index key `<GroupAccount.Address><Group.RowID>`
